File: app/src/main/java/com/alliejc/wcstinder/DancerAdapter.java
package com.alliejc.wcstinder;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.List;


/**
 * Created by acaldwell on 3/18/18.
 */

public class DancerAdapter extends RecyclerView.Adapter<DancerViewHolder> {
    private Context mContext;
    private List<Dancer> mDancerList;
    private IOnSelected mListener;

    public DancerAdapter(Context context, IOnSelected listener) {
        this.mContext = context;
        this.mDancerList = new ArrayList<>();
        this.mListener = listener;
    }

    @NonNull
    @Override
    public DancerViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View view = inflater.inflate(R.layout.item_dancer, parent, false);
        return new DancerViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull DancerViewHolder holder, int position) {
        Dancer dancer = mDancerList.get(holder.getAdapterPosition());
        holder.onBind(mContext, dancer, mListener);
    }

    @Override
    public int getItemCount() {
        return mDancerList.size();
    }

    public void updateAdapter(List list){
        mDancerList = list;
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/alliejc/wcstinder/adapter/DancerAdapter.kt
package com.alliejc.wcstinder.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.ViewGroup
import com.alliejc.wcstinder.R
import com.alliejc.wcstinder.callback.IOnSelected
import com.alliejc.wcstinder.viewholder.DancerViewHolder
import com.alliejc.wcstinder.ext.inflate
import com.alliejc.wcstinder.trackmyswing.Dancer

class DancerAdapter(context: Context, listener:IOnSelected): RecyclerView.Adapter<DancerViewHolder>() {

    var mDancerList: MutableList<Dancer>? = mutableListOf()
    var mContext = context
    var mListener = listener

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): DancerViewHolder {
        var view = parent.inflate(R.layout.item_dancer, false)
        return DancerViewHolder(view)
    }

    override fun onBindViewHolder(holder: DancerViewHolder, position: Int) {
        var dancer = mDancerList?.get(holder.adapterPosition)
        holder.onBind(mContext, dancer, mListener)
    }

    override fun getItemCount(): Int {
        var count = 0
        if(mDancerList != null){
            count = mDancerList!!.size
        }
        return count
    }

    fun updateAdapter(dancer: Dancer){
        mDancerList?.add(dancer)
        notifyDataSetChanged()
    }

    fun updateAdapter(list: MutableList<Dancer>?){
        mDancerList = list
        notifyDataSetChanged()
    }
}